testing/ui-tests/src/utils/pallet_metadata_test_runner.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use codec::{Decode, Encode};
+use frame_metadata::{RuntimeMetadataPrefixed, RuntimeMetadataV14};
+use std::io::Read;
+use subxt_metadata::retain_metadata_pallets;
+
+static TEST_DIR_PREFIX: &str = "subxt_generated_pallets_ui_tests_";
```

Comment:
I'm wondering; build.rs scripts are given an OUT_DIR env var to put things, which is nicely out of the way. I did the same as you with the `metadata_test_runner.rs` stuff, so potentially both could be improved. Def a nice to have though; not needed :)